### PR TITLE
feat(design/autoware-architecture/perception): add radar perception document

### PR DESCRIPTION
**Pull request by scepter914**
## Description

This PR add radar perception document.
The contents of the documents are as below.

perception/supported-functions/radar-based-3d-detector/radar-based-3d-detector

Related PR is https://github.com/autowarefoundation/autoware-documentation/pull/518.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/design/autoware-architecture/perception/reference-implementations/radar-based-3d-detector/faraway-object-detection.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Radar faraway dynamic objects detection with radar objects
+
+## Overview
+
+This diagram describes the pipeline for radar faraway dynamic object detection.
+
+![faraway object detection](image/faraway-object-detection.drawio.svg)
+
+## Supported function
```
Comment (miursh) [inline docs/design/autoware-architecture/perception/reference-implementations/radar-based-3d-detector/faraway-object-detection.md:9]:
```suggestion
## Reference implementation
```

**Review by miursh — COMMENTED** [review]
Just a tiny comment about grammar.

**Review by miursh — APPROVED** [review]
LGTM

**Review by YoshiRi — APPROVED** [review]
LGTM

**Pull request merged**